authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt

```diff
@@ -0,0 +1,195 @@
+            bridgeService = AuthenticatorBridgeServiceSafetyWrapper(
+                IAuthenticatorBridgeService.Stub.asInterface(service)
+            )
+
+            // TODO: Add check for version mismatch between client and server SDKs:
+            // TODO: https://livefront.atlassian.net/browse/BITAU-72
+
+            // Ensure we are using the correct symmetric key:
+            val localKeyFingerprint =
+                symmetricKeyStorageProvider.symmetricKey?.toFingerprint()?.getOrNull()
+
+            // Query bridge service to see if we have a matching symmetric key:
+            val haveCorrectKey = bridgeService
+                .safeCall { checkSymmetricEncryptionKeyFingerprint(localKeyFingerprint) }
+                ?: false
+
+            if (!haveCorrectKey) {
+                // If we don't have the correct key, query for key:
+                symmetricKeyStorageProvider.symmetricKey =
+                    bridgeService.safeCall { symmetricEncryptionKeyData }
+            }
+
+            if (symmetricKeyStorageProvider.symmetricKey == null) {
+                // This means bridgeService returned a null key, which means we can make
+                // no valid operations. We should disconnect form the service and expose to the
+                // calling application that authenticator sync is not enabled.
+                mutableSharedAccountsStateFlow.value = AccountSyncState.SyncNotEnabled
+                unBindService()
+            }
+
+            // Register callback:
+            bridgeService.safeCall { registerBridgeServiceCallback(authenticatorBridgeCallback) }
+
+            // Sync data:
+            bridgeService.safeCall { syncAccounts() }
+        }
+
+        override fun onServiceDisconnected(name: ComponentName) {
+            bridgeService = AuthenticatorBridgeServiceSafetyWrapper(service = null)
+        }
+    }
+
+    override val accountSyncStateFlow: StateFlow<AccountSyncState> =
+        mutableSharedAccountsStateFlow.asStateFlow()
+
+    init {
+        // Listen for lifecycle events
+        processLifecycleOwner.lifecycle.addObserver(
+            object : DefaultLifecycleObserver {
+                override fun onStart(owner: LifecycleOwner) {
+                    bindService()
+                }
+
+                override fun onStop(owner: LifecycleOwner) {
+                    unBindService()
+                }
+            },
+        )
+    }
+
+    private fun bindService() {
+        val intent = Intent().apply {
+            component = ComponentName(
+                connectionType.toPackageName(),
+                AUTHENTICATOR_BRIDGE_SERVICE_CLASS,
+            )
+        }
+
+        isBound = try {
+            applicationContext.bindService(
+                intent,
+                bridgeServiceConnection,
+                Context.BIND_AUTO_CREATE,
+            )
+        } catch (e: SecurityException) {
```

Comment:
You should make sure this is the _only_ exception that can be thrown here. I don't know if there are others. So you'll want to either investigate the docs/source or play around with it.

Reply:
We chatted offline! Based on the docs and looking at few other examples of service binding code, we decided this is sufficient. 

Also, I tested putting "banana" in for package name and the function did not throw.